codebuild/bin/grep_simple_mistakes.sh

```diff
@@ -14,9 +14,48 @@
 
 FAILED=0
 
+#############################################
+# Grep for any instances of raw memcmp() function. s2n code should instead be
+# using s2n_constant_time_equals()
+#
+# KNOWN_MEMCMP_USAGE is used to capture all known uses of memcmp and acts as a
+# safeguard against any new uses of memcmp.
+#############################################
+S2N_FILES_ASSERT_NOT_USING_MEMCMP=$(find "$PWD" -type f -name "s2n*.[ch]" -not -path "*/tests/*")
```

Comment:
consider adding `-not -path "*/bindings/*"` as this will fail if the bindings have been generated.